File: MovieProject/MovieProject.Web/Modules/Northwind/Orders/OrdersColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace MovieProject.Northwind.Columns;

[ColumnsScript("Northwind.Orders")]
[BasedOnRow(typeof(OrdersRow), CheckNames = true)]
public class OrdersColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int OrderId { get; set; }
    [EditLink]
    public string CustomerCompanyName { get; set; }
    public string EmployeeFullName { get; set; }
    public DateTime OrderDate { get; set; }
    public DateTime RequiredDate { get; set; }
    public DateTime ShippedDate { get; set; }
    public string ShipViaCompanyName { get; set; }
    public decimal Freight { get; set; }
    public string ShipName { get; set; }
    public string ShipAddress { get; set; }
    public string ShipCity { get; set; }
    public string ShipRegion { get; set; }
    public string ShipPostalCode { get; set; }
    public string ShipCountry { get; set; }
}

File: MovieProject/MovieProject.Web/Modules/ServerTypes/Northwind/OrdersColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrdersRow } from "./OrdersRow";

export interface OrdersColumns {
    OrderId: Column<OrdersRow>;
    CustomerCompanyName: Column<OrdersRow>;
    EmployeeFullName: Column<OrdersRow>;
    OrderDate: Column<OrdersRow>;
    RequiredDate: Column<OrdersRow>;
    ShippedDate: Column<OrdersRow>;
    ShipViaCompanyName: Column<OrdersRow>;
    Freight: Column<OrdersRow>;
    ShipName: Column<OrdersRow>;
    ShipAddress: Column<OrdersRow>;
    ShipCity: Column<OrdersRow>;
    ShipRegion: Column<OrdersRow>;
    ShipPostalCode: Column<OrdersRow>;
    ShipCountry: Column<OrdersRow>;
}

export class OrdersColumns extends ColumnsBase<OrdersRow> {
    static readonly columnsKey = 'Northwind.Orders';
    static readonly Fields = fieldsProxy<OrdersColumns>();
}